### ISSUE TITLE
AWS SQS fifo queue

### ISSUE THREAD

**Issue by davidedonattini** [uecode/qpush-bundle#116]
Hi,
I try to publish a message on a SQS fifo queue but I get this error:

[Aws\Sqs\Exception\SqsException]                                                                                                                                                                                                                                                                  
  Error executing "SendMessage" on "https://sqs.us-east-2.amazonaws.com/629839571894/my_queue.fifo"; AWS HTTP error: Client error: `POST https://sqs.us-east-2.amazonaws.com/629839571894/my_queue.fifo` resulted in a `400 Bad Request` response:                          
  <?xml version="1.0"?><ErrorResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/"><Type>Sender</Type><Code>M (truncated...)                                                                                                                                                                  
   MissingParameter (client): The request must contain the parameter MessageGroupId. - <?xml version="1.0"?><ErrorResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/"><Type>Sender</Type><Code>MissingParameter</Code><Message>The request must contain the parameter MessageGroupId.</Mes  
  sage><Detail/><RequestId>88a24682-1133-5b92-9718-1674a1b28a92</RequestId></ErrorResponse>

I think that missed the MessageGroupId from the request.

**Comment by c0ntax** [comment]
You can fix that by hacking the following into AwsProvider.php

```php
        $result = $this->sqs->sendMessage([
            'MessageGroupId' => 1,
            'MessageDeduplicationId' => md5(json_encode($message)),
            'QueueUrl'      => $this->queueUrl,
            'MessageBody'   => json_encode($message),
            'DelaySeconds'  => $options['message_delay']
        ]);
```

I've hit the same problem. I'm going to make a custom Provider that extends the AwsProvider and implements it's own publish() method that takes a MessageGroupId and MessageDeduplicationId in the $option[] array.

**Comment by kmfk** [comment]
This should now be resolved by the addition of #120
